java/ql/src/experimental/Security/CWE/CWE-094/ScriptInjection.ql

```diff
@@ -14,31 +14,65 @@ import java
 import semmle.code.java.dataflow.FlowSources
 import DataFlow::PathGraph
 
+/** A method of ScriptEngine that allows code injection. */
 class ScriptEngineMethod extends Method {
   ScriptEngineMethod() {
     this.getDeclaringType().getASupertype*().hasQualifiedName("javax.script", "ScriptEngine") and
     this.hasName("eval")
+    or
+    this.getDeclaringType().getASupertype*().hasQualifiedName("javax.script", "Compilable") and
+    this.hasName("compile")
+    or
+    this.getDeclaringType().getASupertype*().hasQualifiedName("javax.script", "ScriptEngineFactory") and
+    (
+      this.hasName("getProgram") or
+      this.hasName("getMethodCallSyntax")
+    )
```

Comment:
```suggestion
    this.hasName(["getProgram", "getMethodCallSyntax"])
```

Reply:
Thanks @intrigus-lgtm. I've made the change.